src/pyfaf/ureport.py

```diff
@@ -530,6 +532,17 @@ def save_attachment(db, attachment):
     else:
         log.warning("Unknown attachment type")
 
+def valid_known_type(known_type):
```

Comment:
Try adding comments for all of the functions. Still I think 'known'  and 'known_type' do not describe what this all does which is matching of similar ureports.